### DISCUSSION TITLE
Incorrect response example with pydantic or TypedDict in newset version

### DISCUSSION THREAD

**Discussion by ptyin** [fastapi/fastapi#11067] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uuid
from typing import Any

from fastapi import FastAPI
from pydantic import BaseModel
from typing_extensions import TypedDict

app = FastAPI()


class GenerateRsp(BaseModel):
    tag: str


@app.post('a')
async def a() -> TypedDict("GenerateRsp", {"tag": str}):
    tag = uuid.uuid1()
    return {"tag": str(tag)}


@app.post('b', response_model=GenerateRsp)
async def b() -> Any:
    tag = uuid.uuid1()
    return {"tag": str(tag)}
```

### Description

Use the minimum reproducible example above, neither of the two functions get correct example values.
<img width="1244" alt="image" src="https://github.com/tiangolo/fastapi/assets/32592585/d4a79921-18bc-452d-b647-6ae418bdbe8d">

It works fine with version `0.100.0` though (shown as followed), so I believe it is either a bug or a missing updated usage in documentation.
<img width="1242" alt="image" src="https://github.com/tiangolo/fastapi/assets/32592585/e3a736ef-8cc5-40cd-aed9-061595863b69">

### Operating System

macOS

### FastAPI Version

0.109.0

### Pydantic Version

2.6.0

### Python Version

3.9.6

### Additional Context

Tested with 0.100.0 (✅), 0.108.0(❌), 0.109.0(❌)

**Comment by YuriiMotov** [comment]
This was fixed in FastAPI 0.111.1 🎉
